src/mysky/index.ts

```diff
@@ -491,6 +501,105 @@ export class MySky {
     return { data: entry.data };
   }
 
+  // ===============
+  // Encrypted Files
+  // ===============
+
+  /**
+   * Lets you get the share-able path seed, which can be passed to
+   * file.getJSONEncrypted. Requires read permission on the path.
+   *
+   * @param path - The given path.
+   * @param isDirectory - Whether the path is a directory.
+   * @returns - The seed for the path.
+   */
+  async getEncryptedFileSeed(path: string, isDirectory: boolean): Promise<string> {
+    validateString("path", path, "parameter");
+    validateBoolean("isDirectory", isDirectory, "parameter");
+
+    return await this.connector.connection.remoteHandle().call("getEncryptedFileSeed", path, isDirectory);
+  }
```

Comment:
do not await returned value, it's implicitly async anyway and the call will need to be awaited anyway

```suggestion
  getEncryptedFileSeed(path: string, isDirectory: boolean): Promise<string> {
    validateString("path", path, "parameter");
    validateBoolean("isDirectory", isDirectory, "parameter");

    return this.connector.connection.remoteHandle().call("getEncryptedFileSeed", path, isDirectory);
  }
```

Reply:
https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise

Reply:
fwiw I'm in the `return await` camp, but not for the reason Marcin linked. If I remember correctly it has another advantage which is the call stack. If you await the promise before returning it this function will be part of the call stack, which will be useful if an error is thrown.